packages/url_launcher/url_launcher_web/lib/src/link.dart

```diff
@@ -200,19 +209,50 @@ class LinkViewController extends PlatformViewController {
     // The keydown event is not directly associated with the target Link, so
     // we need to look for the recently hit tested Link to handle the event.
     if (_hitTestedViewId != null) {
-      _instances[_hitTestedViewId]?._onDomKeydown();
+      final LinkViewController? instance = _instances[_hitTestedViewId];
+      final int? viewId = instance?.viewId;
+      assert(
+        _hitTestedViewId == viewId,
+        'Keydown event should only be handled by the hit tested Link',
+      );
+      instance?._triggerLink(event: null, isKeyboardEvent: true);
     }
     // After the keyboard event has been received, clean up the hit test state
     // so we can start fresh on the next event.
     unregisterHitTest();
   }
 
+  /// Keeps track of whether a hit test was registered.
+  static bool followLinkTriggered = false;
+
+  /// Keeps track of the last mouse event captured from our global click handler.
+  static html.MouseEvent? triggeredMouseEvent;
+
+  /// Global click handler that triggers on the `capture` phase. We use `capture`
+  /// because some events may be consumed and prevent further propagation at the
+  /// target. This may lead to issues (see: https://github.com/flutter/flutter/issues/143164)
+  /// where a followLink was executed but the event never bubbles back up to the
+  /// window (e.g. when button semantics obscure the platform view). We make sure
+  /// to only trigger the link if a hit test was registered and remains valid at
+  /// the time the click handler executes.
   static void _onGlobalClick(html.MouseEvent event) {
-    final int? viewId = getViewIdFromTarget(event);
-    _instances[viewId]?._onDomClick(event);
-    // After the DOM click event has been received, clean up the hit test state
-    // so we can start fresh on the next event.
-    unregisterHitTest();
+    final html.Element? targetElement = event.target as html.Element?;
+
+    // We only want to cache the mouse event if its target is a link, so that 
+    // we can later prevent its default behavior and handle how it should 
+    // open the link ourselves.
+    if(isLinkElement(targetElement) || isSemanticLinkElement(targetElement)) {
```

Comment:
@mdebbar I had to add this to fix the issue of caching mouse events that aren't relevant to us (since the global handler captures every mouse event).  

I added `isSemanticLinkElement` which just checks if the target element is a semantic node that's wrapped within an `<a>` tag.  Is this robust enough to handle the semantic link case? Or should we try to associate a platform view's `viewId` to its children's generated semantic nodes?